cmd/pdatagen/internal/base_fields.go

```diff
@@ -462,3 +474,89 @@ func (pf *primitiveAsDoubleField) generateCopyToValue(sb *strings.Builder) {
 }
 
 var _ baseField = (*primitiveField)(nil)
+
+type oneOfPrimitiveValue struct {
+	name            string
+	defaultVal      string
+	testVal         string
+	returnType      string
+	originFieldName string
+	originFullName  string
+	fieldType       string
+}
+
+func (opv *oneOfPrimitiveValue) generateAccessors(ms baseStruct, sb *strings.Builder) {
+	sb.WriteString(os.Expand(accessorsOneofPrimitiveTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.getName()
+		case "fieldName":
+			return opv.name
+		case "lowerFieldName":
+			return strings.ToLower(opv.name)
+		case "returnType":
+			return opv.returnType
+		case "originFieldName":
+			return opv.originFieldName
+		case "originFullName":
+			return opv.originFullName
+		case "fieldType":
+			return opv.fieldType
+		default:
+			panic(name)
+		}
+	}))
+	sb.WriteString("\n")
+}
+
+func (opv *oneOfPrimitiveValue) generateAccessorsTest(ms baseStruct, sb *strings.Builder) {
+	sb.WriteString(os.Expand(accessorsPrimitiveTestTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.getName()
+		case "defaultVal":
+			return opv.defaultVal
+		case "fieldName":
+			return opv.name
+		case "testValue":
+			return opv.testVal
+		default:
+			panic(name)
+		}
+	}))
+	sb.WriteString("\n")
+}
+
+type numberField struct {
+	fields []oneOfPrimitiveValue
+}
+
+func (nf *numberField) generateAccessors(ms baseStruct, sb *strings.Builder) {
+	for _, field := range nf.fields {
+		field.generateAccessors(ms, sb)
+	}
+}
+
+func (nf *numberField) generateAccessorsTest(ms baseStruct, sb *strings.Builder) {
+	for _, field := range nf.fields {
+		field.generateAccessorsTest(ms, sb)
+	}
+}
+
+func (nf *numberField) generateSetWithTestValue(sb *strings.Builder) {
+	for _, field := range nf.fields {
+		sb.WriteString("\ttv.Set" + field.name + "(" + field.testVal + ")\n")
```

Comment:
This line can be in a func "func (opv *oneOfPrimitiveValue) generateSetWithTestValue(sb *strings.Builder)" so we can make smarter things later :)

Reply:
Done!